lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -3261,6 +3261,12 @@ class ExprAvailabilityWalker : public ASTWalker {
         // Regex literals require both the Regex<Output> type to be available,
         // as well as the initializer that is implicitly called.
         diagnoseDeclRefAvailability(Context.getRegexDecl(), Range);
+      } else if (auto literalType = LE->getType()) {
```

Comment:
Instead of making this an `else if`, I think we can just remove the regex case here as a regex literal is always typed as `Regex<Output>`.